src/main/core/DependencyInjection/Compiler/MessengerConfigPass.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Claroline Connect package.
+ *
+ * (c) Claroline Consortium <consortium@claroline.net>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Claroline\CoreBundle\DependencyInjection\Compiler;
+
+use Claroline\CoreBundle\Library\Configuration\PlatformConfigurationHandler;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class MessengerConfigPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        /** @var PlatformConfigurationHandler $platformConfig */
+        $platformConfig = $container->get(PlatformConfigurationHandler::class);
+
+        if ($platformConfig->getParameter('job_queue.enabled')) {
+            switch ($platformConfig->getParameter('job_queue.transport')) {
+                case 'doctrine':
+                    $platformConfig->setParameter('job_queue.dsn', 'doctrine://default');
+                    break;
+                case 'redis':
+                    $platformConfig->setParameter(
+                        'job_queue.dsn',
+                        'redis://password@localhost:6379?auto_setup=true&dbindex=1'
+                    );
+                    break;
+            }
+        } else {
+            $platformConfig->setParameter('job_queue.dsn', 'sync://');
```

Comment:
As you pointed out privately, this won't have the intended effect because we are currently configuring the dsn from an env var. 
Changing the value of an env var is a no-go at this stage, since it means accessing globals, which should only be done from config files at bootstrap time for a matter of good practices.

Here is how we can redefine the DSN:
```php
if (!$container->hasDefinition('messenger.transport.default')) {
    throw new \LogicException('The symfony/messenger seems misconfigured: the "default" transport does not exist.);
}

$dsn = 'sync://';

if ($platformConfig->getParameter('job_queue.enabled')) {
    $dsn = '.......';
    // ...
}

$container->getDefinition('messenger.transport.default')->replaceArgument(0, $dsn);
```
Another alternative could be to use a regular container parameter instead of a env parameter.
Which means changing the suggested config to something like:

```yaml
messenger:
    # ...
    dsn: %claroline.messenger.dsn%`
```

And just changing the value of the parameter from this compiler pass.

The latter is less coupled to symfony internals so it might be the best one. 
@Elorfin opinion welcome :)

Reply:
Using a param `messenger.dsn` seems good to me.